File: AndroidJava_Hahow/Currency/app/src/main/java/com/codingbydumbbell/homework_currency/MainActivity.java
package com.codingbydumbbell.homework_currency;

import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private float exchangeRatio_NTDUSD = 30.9f;
    private EditText edNtd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViews();
    }

    private void findViews() {

        edNtd = findViewById(R.id.ntd);
    }

    public void go(View view) {

        String ntd = edNtd.getText().toString();

        if (!ntd.isEmpty()) {

            String usd = String.format("%.4f", Float.parseFloat(ntd) / exchangeRatio_NTDUSD);

            new AlertDialog.Builder(this)
                    .setTitle(R.string.result)
                    .setMessage(getString(R.string.usd_is) + usd)
                    .setPositiveButton(R.string.ok, null)
                    .show();
        } else {

            new AlertDialog.Builder(this)
                    .setTitle(R.string.problem)
                    .setMessage(R.string.please_enter_ntd)
                    .setPositiveButton(R.string.ok, null)
                    .show();
        }
    }
}

File: AndroidJava_Hahow/BMI_Kotlin/app/src/main/java/com/codingbydumbbell/app_bmik/MainActivity.kt
package com.codingbydumbbell.app_bmik

import android.content.DialogInterface
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        help.setOnClickListener {
            AlertDialog.Builder(this@MainActivity)
                    .setMessage("The body mass index (BMI) or Quetelet index is a value derived from the mass (weight) and height of an individual")
                    .setPositiveButton("OK") { dialog: DialogInterface?, which: Int ->
                        Toast.makeText(this, "Testing", Toast.LENGTH_LONG).show()
                    }.show()
        }
        bmi.setOnClickListener {
            val w = weight.text.toString().toFloat()
            val h = height.text.toString().toFloat()
            val bmi = w / (h * h)
            val intent = Intent(this@MainActivity, ResultActivity::class.java)
            intent.putExtra("BMI", bmi)
            startActivity(intent)
        }
    }
}
